layersvt/api_dump.h

```diff
@@ -55,30 +55,69 @@ enum class ApiDumpFormat {
 class ApiDumpSettings {
    public:
     ApiDumpSettings() {
-        // Get the output file settings and create a stream for it
+        std::string filename_string = "";
+        // If the layer settings file has a flag indicating to output to a file,
+        // do so, to the appropriate filename.
         const char *file_option = getLayerOption("lunarg_api_dump.file");
         if (file_option != NULL && strcmp(file_option, "TRUE") == 0) {
-            use_cout = false;
             const char *filename_option = getLayerOption("lunarg_api_dump.log_filename");
             if (filename_option != NULL && strcmp(filename_option, "") != 0) {
-                output_stream.open(filename_option, std::ofstream::out | std::ostream::trunc);
-                size_t last_slash_idx = std::string(filename_option).find_last_of("\\/");
-                if (std::string::npos != last_slash_idx) {
-                    output_dir = std::string(filename_option).substr(0, last_slash_idx + 1);
-                }
+                filename_string = filename_option;
             } else {
-                output_stream.open("vk_apidump.txt", std::ofstream::out | std::ostream::trunc);
+                filename_string = "vk_apidump.txt";
+            }
+        }
+        // If an environment variable is set, always output to that filename instead,
+        // whether or not the settings file enables the option.  Just assume a non-empty
+        // string is asking for the file output to the given name.
+        std::string env_value = GetPlatformEnvVar("APIDUMP_LOG_FILENAME");
+        if (!env_value.empty()) {
+            filename_string = env_value;
+        }
+        // If one of the above has set a filename, open the file as an output stream.
+        if (!filename_string.empty()) {
+            use_cout = false;
+            output_stream.open(filename_string, std::ofstream::out | std::ostream::trunc);
+            size_t last_slash_idx = filename_string.find_last_of("\\/");
```

Comment:
What about  "\\" used to in Windows path names?

Reply:
This looks for the last of either "\" or "/"  the "\\" translates to "\".

Reply:
Doh! Misread the code.